src/reformat.c

```diff
@@ -116,6 +117,8 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
         state->unormFloatTableY[cp] = (float)unormY / state->yuvMaxChannelF;
         if (state->mode == AVIF_REFORMAT_MODE_IDENTITY) {
             state->unormFloatTableUV[cp] = (float)unormY / state->yuvMaxChannelF;
+        } else if (state->mode == AVIF_REFORMAT_MODE_YCGCO) {
+            state->unormFloatTableUV[cp] = ((float)unormY - state->uvBias) / state->yuvMaxChannelF;
```

Comment:
This line feels wrong. I feel like it should either be the math from the above block or the below block, and somehow it is something in between. Is this correct?

Reply:
You're right. For YCgCo, unormFloatTableY and unormFloatTableUV should be applied at the end of the YUV-to-RGB conversion. Let me look into this.

Reply:
I think you should just avoid `uvBias` the way IDENTITY does, in both `avifPrepareReformatState()` and `avifReformatStateUVToUNorm()`.

Reply:
Something like this, perhaps?

    diff --git a/src/reformat.c b/src/reformat.c
    index 33c0f7b..de550c2 100644
    --- a/src/reformat.c
    +++ b/src/reformat.c
    @@ -115,10 +115,8 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
                 unormUV = avifLimitedToFullUV(image->depth, unormUV);
             }
             state->unormFloatTableY[cp] = (float)unormY / state->yuvMaxChannelF;
    -        if (state->mode == AVIF_REFORMAT_MODE_IDENTITY) {
    +        if ((state->mode == AVIF_REFORMAT_MODE_IDENTITY) || (state->mode == AVIF_REFORMAT_MODE_YCGCO)) {
                 state->unormFloatTableUV[cp] = (float)unormY / state->yuvMaxChannelF;
    -        } else if (state->mode == AVIF_REFORMAT_MODE_YCGCO) {
    -            state->unormFloatTableUV[cp] = ((float)unormY - state->uvBias) / state->yuvMaxChannelF;
             } else {
                 state->unormFloatTableUV[cp] = ((float)unormUV - state->uvBias) / state->yuvMaxChannelF;
             }
    @@ -139,7 +137,7 @@ static int avifReformatStateYToUNorm(avifReformatState * state, float v)
     static int avifReformatStateUVToUNorm(avifReformatState * state, float v)
     {
         int unorm = (int)avifRoundf(v * state->yuvMaxChannelF);
    -    if (state->mode != AVIF_REFORMAT_MODE_IDENTITY) { // Don't adjust U or V when data is raw RGB
    +    if (state->mode == AVIF_REFORMAT_MODE_YUV_COEFFICIENTS) { // Don't adjust U or V when data is raw RGB or YCgCo
             unorm += state->uvBias;
         }
         unorm = AVIF_CLAMP(unorm, 0, state->yuvMaxChannel);

Reply:
I could probably even flip the conditional in the `avifPrepareReformatState()` to match the one in `avifReformatStateUVToUNorm()` and simplify it.

Reply:
Ignore that diff; I've done something wrong.

Reply:
Okay, upon further reading of the CICP formulas, I think your usage of `uvBias` is _correct_. The only open question is whether or not to use the limited Y or limited UV range. Your original code might have been correct this whole time. Sorry for the swirl, if so.

Reply:
Yeah, the line above equation (20) makes it quite clear. Your current implementation is correct. Sorry about that.